packages/labs/signals/src/lib/watch.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {DirectiveResult, Part, directive} from 'lit/directive.js';
+import {AsyncDirective} from 'lit/async-directive.js';
+import {Signal} from 'signal-polyfill';
+import {SignalWatcher} from './signal-watcher.js';
+
+export class WatchDirective<T> extends AsyncDirective {
+  private __host?: SignalWatcher;
+
+  private __signal?: Signal.State<T> | Signal.Computed<T>;
+
+  private __watcher?: Signal.subtle.Watcher;
+
+  // We have to wrap the signal in a computed to work around a bug in the
+  // signal-polyfill: https://github.com/proposal-signals/signal-polyfill/issues/27
+  private __computed?: Signal.Computed<T | undefined>;
+
+  private __watch() {
+    if (this.__watcher === undefined) {
+      this.__computed = new Signal.Computed(() => {
+        return this.__signal?.get();
+      });
+      const watcher = (this.__watcher = new Signal.subtle.Watcher(() => {
+        this.__host?.updateWatchDirective(this as WatchDirective<unknown>);
+        watcher.watch();
+      }));
+      watcher.watch(this.__computed);
+    }
+  }
+
+  private __unwatch() {
+    if (this.__watcher !== undefined) {
+      this.__watcher.unwatch(this.__computed!);
+      this.__computed = undefined;
+      this.__watcher = undefined;
+    }
+  }
+
+  commmit() {
+    this.setValue(Signal.subtle.untrack(() => this.__computed?.get()));
+  }
+
+  render(signal: Signal.State<T> | Signal.Computed<T>): T {
+    // This would only be called if render is called directly, like in SSR.
+    return Signal.subtle.untrack(() => signal.get());
+  }
+
+  override update(
+    part: Part,
+    [signal]: [signal: Signal.State<T> | Signal.Computed<T>]
+  ) {
+    this.__host ??= part.options?.host as SignalWatcher;
+    if (signal !== this.__signal && this.__signal !== undefined) {
+      // Unwatch the old signal
+      this.__unwatch();
+    }
+    this.__signal = signal;
+    this.__watch();
+
+    // We use untrack() so that the signal access is not tracked by the watcher
+    // created by SignalWatcher. This means that an can use both SignalWatcher
+    // and watch() and a signal update won't trigger a full element update if
+    // it's only passed to watch() and not otherwise accessed by the element.
+    return Signal.subtle.untrack(() => this.__computed!.get());
+  }
+
+  protected override disconnected(): void {
+    this.__unwatch();
+  }
+
+  protected override reconnected(): void {
+    this.__watch();
+  }
+}
+
+export type WatchDirectiveFunction = <T>(
+  signal: Signal.State<T> | Signal.Computed<T>
+) => DirectiveResult<typeof WatchDirective<T>>;
+
+/**
+ * Renders a signal and subscribes to it, updating the part when the signal
+ * changes.
+ *
+ * watch() can only be used in a reactive element that applies the
```

Comment:
It would be nice if this could work without reactive element. Considering how to facilitate that:
1. we could have `update` not use `untrack` if there's no host and then have the watcher call `commit` in a queued microtask (to avoid the read exception).
2. OR create a synthetic host that implements the `updateWatchDirective` protocol in a microtask, perhaps keyed via a WeakMap using the `part.options` (if that's always available?).

Reply:
+1 to supporting lit-html + signals outside of ReactiveElement for lit-html standalone users. Could possibly lead to powerful server-powered template renderers

Reply:
Via chat, we resolved not to block on adding this but instead to add a comment stating this will be added in the future based on feedback.

Reply:
I added a comment to `_updateWatchDirective()` suggesting how we might move forward on this.